frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUViewImpl.java

```diff
@@ -173,6 +176,7 @@ private VerticalLayout buildMainLayout() {
         buttonCreateDPU.setHeight("25px");
         buttonCreateDPU.setWidth("150px");
         buttonCreateDPU.addStyleName("v-button-primary");
+        buttonCreateDPU.setVisible(utils.hasUserAuthority("dpuTemplate.create"));
```

Comment:
This approach using utils.hasUserAuthority("dpuTemplate.create") is redundant to the standard approach: 

@Autowired
private AuthAwarePermissionEvaluator permissions;

permissions.hasPermission(pipeline, "pipeline.create");

which is used in most of the situations where permission should be checked. 

Solution: Do not use Utils class to check permissions. Utils should not contain any auth/autorization methods. Use ALWAYS one class, the PermissionEvaluator class!